src/main.c

```diff
@@ -3440,18 +3440,31 @@ int main(int argc, char **argv)
         conf.error_log = NULL;
     }
 
-    { /* raise RLIMIT_NOFILE */
-        struct rlimit limit;
+    {
+        struct rlimit limit = { 0 };
```

Comment:
Am I correct to assume that this change is to make sure that `limit.rlim_cur` is initialized in case `getrlimit` fails? If so, I'm not sure if this is the best thing to do.

If we are to think that `getrlimit` might fail (as current code does), I do not think it's correct to report a fatal error when it does fail.
If we are to think that `getrlimit` never fails, then we should raise an fatal error indicating that it failed. The error message should indicate exactly that. It should also be calling `h2o_fatal` instead of returning EX_CONFIG, as the presumption would be that getrlimit`` never fails.

Regarding which path to take, I think my weak preference goes to the latter, as it would be less prone to errors.

Reply:
@kazuho This change is not specifically to deal with `getrlimit()`.  That logic was not changed.  The main point of this patch is to make sure that the configured `max-connections` value does not exceed `RLIMIT_NOFILE` of the process.

Reply:
I think I understand the intention.

My point is that the code with proposed changes is written with the assumption that `getrlimit` might fail (hence the entire `if` branch based on the return value of `getrlimit`), but fails to let the server start up when `getrlimit` fails.

Reply:
I see your point.  I just added the check independently of the failure status of `getrlimit()`.  I think your point is we should let the server start up even if `getrlimit()` fails.  Would it be better to just log a warning and continue instead of fatal and exit in my new check for `conf.max_connections > (int)limit.rlim_cur`?  Or should my check only happen if `getrlimit()` succeeds?

Reply:
This is nothing more than a humble opinion and I'm not sure if works, but how about adding the check right after `getrlimit` returns success? At that point, we can check against `rlim_max` (i.e., hard limit), and if that is above `max_connections` raise an error.

Reply:
Thanks @kazuho.  This feedback has been incorporated in [79b85e46](https://github.com/h2o/h2o/pull/2705/commits/79b85e46e0a11e7697ea8c8c0f34716d29f59bea).